src/rpcblockchain.cpp

```diff
@@ -51,6 +51,32 @@ double GetDifficulty(const CBlockIndex* blockindex)
     return dDiff;
 }
 
+UniValue blockheaderToJSON(const CBlockIndex* blockindex)
```

Comment:
There's a strong overlap with #5486 here - even the function is called the same. This does expose a few more values, also derived ones like `confirmation` and `chainwork`.